### ISSUE TITLE
Scrollable Checkboxes label not read by screen reader

### ISSUE THREAD

**Issue by avernet** [orbeon/orbeon-forms#4733]
[+1 from customer](https://3.basecamp.com/3600924/buckets/1966555/messages/3259104091)

**Comment by avernet** [comment]
Out-of-the-box, the label is rendered as a `<span>` with no `id`, and the group has no `aria-labelledby`:

<img width="710" alt="70239DC4-BA4E-4E3B-8B42-0D17D91C6194" src="https://user-images.githubusercontent.com/106961/102856104-bc89a180-43da-11eb-8443-2161244ee340.png">

Adding `xxbl:label-for="select"`, the label is now a `<label>` instead of a `<span>` (good), the group got an `aria-labelledby`, but it doesn't point to a valid id, and the `<label>` has not id:

<img width="716" alt="13A6ABB8-3B10-4B63-ABDF-B6A2BDD17120" src="https://user-images.githubusercontent.com/106961/102856147-ca3f2700-43da-11eb-8efb-68294cc0af78.png">

**Comment by avernet** [comment]
Made the change to the XBL component, added @ebruchez who needs to fix the issue shown in the second screenshot of my previous comment.

**Comment by avernet** [comment]
This already fails even without XBL; say you have:

```xml
<xf:label for="control">Please enter your first name:</xf:label>
<xf:select ref="." appearance="full" id="control">
    <xf:item>
        <xf:value>true</xf:value>
        <xf:label>Approve</xf:label>
    </xf:item>
</xf:select>
```

The generated HTML is as follows. Note the `aria-labelledby="xf-4"`, while the label has `id="control≡≡l"` (and there nothing in the page with id `xf-4`).

```html
<label class="xforms-label" id="control≡≡l">Please enter your first name:</label>
<span id="control" class="xforms-control xforms-select xforms-incremental xforms-select-appearance-full">
    <span class="xforms-items xforms-help-popover-control" role="group" aria-labelledby="xf-4">
        <span class="xforms-deselected">
            <label class="checkbox">
                <input id="control≡≡e0" type="checkbox" name="control" value="0">
                <span>Approve</span>
            </label>
        </span>
    </span>
</span>
```

**Comment by avernet** [comment]
This got fixed when we fixed #4831.
